core/include/detray/masks/masks.hpp

```diff
@@ -172,6 +177,17 @@ class mask {
         return ss.str();
     }
 
+    template <size_type parameter_dim>
```

Comment:
Just an idea: Can we not template this function on the ```matrix_actor``` instead of the class? Then the masks don't have to compile the ```matrix_actor``` all the time and we will also not see that type in error messages for the mask store etc? Because the matrix types are very verbose

Also, I guess we can use different matrices with the same mask object, I guess

Reply:
Hmm maybe not.. if we do that we would need to always specify the matrix type whenever we use the function.
And does it really reduce the error messages even when the mask has the template parameter of `transform3` object?

Reply:
Oh, sorry, I did not look at the code closely enough. Somehow I thought you had templated the entire mask class on the matrix actor. Don't know how I got that idea, so, never mind this comment :D